crates/isograph_schema/src/isograph_schema.rs

```diff
@@ -191,7 +198,9 @@ impl<TSchemaValidationState: SchemaValidationState> Schema<TSchemaValidationStat
         TSchemaValidationState::ClientFieldSelectionLinkedFieldAssociatedData,
         TSchemaValidationState::VariableDefinitionInnerType,
     > {
-        &self.client_fields[client_field_id.as_usize()]
+        match &self.client_fields[client_field_id.as_usize()] {
+            ClientType::ClientField(client_field) => client_field,
+        }
```

Comment:
This gives me a bit of pause, but I suppose it's fine to panic here if the invariant is violated (i.e. the client field ID refers to a pointer)